lib/graphql/language/nodes.rb

```diff
@@ -73,7 +73,7 @@ def position
         end
 
         def to_query_string(printer: GraphQL::Language::Printer.new)
-          printer.print(self)
+          @query_string ||= printer.print(self)
```

Comment:
what if a different `printer:` is passed in here? 

I think we could get the lion's share of the benefit with something like 

```ruby
if printer.class = GraphQL::Language::Printer 
  @query_string ||= ... 
else 
  ... 
end 
```

does that seem appropriate?

Reply:
Yes, that makes sense!
But it's also hard to reason why a different printer would be returning anything different if an AST node has not changed.

Reply:
Oh, now I realize what the problem is:
Some could pass different instances of printer to this method